### PR TITLE
fix magnetic field calculation based on geo lookup for WMM2018

### PR DESCRIPTION
**Pull request by EliaTarasov**
Relates to [issue](https://github.com/PX4/sitl_gazebo/issues/278).

### CODE REVIEW

**Review by TSC21 — APPROVED** [review]

**Comment by EliaTarasov** [comment]
Zurich [master](https://review.px4.io/plot_app?log=45c0d5ac-ce0e-480c-b80e-8238715ce1d2)
Zurich [pr](https://review.px4.io/plot_app?log=05e54183-80ca-4947-b50c-d15c2b86f74d)

File: src/geo_mag_declination.cpp [hunk 1]
```diff
@@ -34,9 +34,8 @@
 /**
 * @file geo_mag_declination.c
```
Comment (julianoes) [inline src/geo_mag_declination.cpp]: While you're at it, could you fix the comment for the filename?

File: src/gazebo_mavlink_interface.cpp [hunk 2]
```diff
@@ -593,10 +582,23 @@ void GazeboMavlinkInterface::SendSensorMessages()
 #endif
   ignition::math::Vector3d pos_n = q_ng.RotateVector(pos_g);
 
-  float declination = get_mag_declination(groundtruth_lat_rad, groundtruth_lon_rad);
+  // Magnetic declination and inclination (radians)
+  float declination = get_mag_declination(groundtruth_lat_rad * 180 / M_PI, groundtruth_lon_rad * 180 / M_PI) * M_PI / 180;
+  float inclination = get_mag_inclination(groundtruth_lat_rad * 180 / M_PI, groundtruth_lon_rad * 180 / M_PI) * M_PI / 180;
+
+  // Magnetic strength (10^5xnanoTesla)
+  float strength = 0.01f * get_mag_strength(groundtruth_lat_rad * 180 / M_PI, groundtruth_lon_rad * 180 / M_PI);
```
Comment (julianoes) [inline src/gazebo_mavlink_interface.cpp]: Nitpick: my taste is to add the unit  if possible, e.g. `strength_ga`.

File: src/geo_mag_declination.cpp [hunk 3]
```diff
@@ -34,9 +34,8 @@
...
 *
-* Calculation / lookup table for earth magnetic field declination.
-*
-* Lookup table from Scott Ferguson <scottfromscott@gmail.com>
+* Calculation / lookup table for Earth's magnetic field declination (deg), inclination (deg) and strength (mTesla).
+* Data generated by https://www.ngdc.noaa.gov/geomag-web/#igrfgrid IGRF calculator on 22 Jan 2018
```
Comment (julianoes) [inline src/geo_mag_declination.cpp:38]: :+1:

**Review by julianoes — APPROVED** [review]
Thanks, looks correct from what I can tell.

File: src/gazebo_mavlink_interface.cpp (hunk 2 above)
Comment (EliaTarasov) [reply src/gazebo_mavlink_interface.cpp]: Done.

File: src/geo_mag_declination.cpp (hunk 1 above)
Comment (EliaTarasov) [reply src/geo_mag_declination.cpp]: Done.

**Review by julianoes — APPROVED** [review]
Thx!

File: src/geo_mag_declination.cpp (hunk 3 above)
Comment (CarlOlsson) [reply src/geo_mag_declination.cpp:38]: > Calculation / lookup table for Earth's magnetic field declination (deg), inclination (deg) and strength (mTesla).

the strength table is in centi-Tesla ;)

Comment (EliaTarasov) [reply src/geo_mag_declination.cpp:38]: Thx, fixed.

**Comment by EliaTarasov** [comment]
@TSC21 @julianoes @CarlOlsson can we proceed?

**Comment by julianoes** [comment]
Thanks @EliaTarasov!

**Pull request merged**